onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -208,6 +209,44 @@ MIGraphXExecutionProvider::MIGraphXExecutionProvider(const MIGraphXExecutionProv
 MIGraphXExecutionProvider::~MIGraphXExecutionProvider() {
 }
 
+AllocatorPtr MIGraphXExecutionProvider::CreateMIGraphXAllocator(OrtDevice::DeviceId device_id,
+                                                        size_t migx_mem_limit,
+                                                        ArenaExtendStrategy arena_extend_strategy,
+                                                        MIGraphXExecutionProviderExternalAllocatorInfo
+                                                            external_allocator_info,
+                                                        const OrtArenaCfg* default_memory_arena_cfg) {
+  if (external_allocator_info.UseExternalAllocator()) {
+    AllocatorCreationInfo default_memory_info(
+        [external_allocator_info](OrtDevice::DeviceId id) {
+          return std::make_unique<MIGraphXExternalAllocator>(id, HIP,
+                                                         external_allocator_info.alloc,
+                                                         external_allocator_info.free,
+                                                         external_allocator_info.empty_cache);
+        },
+        device_id,
+        false);
+
+    return CreateAllocator(default_memory_info);
+  } else {
+    AllocatorCreationInfo default_memory_info(
+        [](OrtDevice::DeviceId id) {
+          return std::make_unique<MIGraphXAllocator>(id, HIP);
+        },
+        device_id,
+        true,
+        {default_memory_arena_cfg ? *default_memory_arena_cfg
+                                  : OrtArenaCfg(migx_mem_limit, static_cast<int>(arena_extend_strategy),
+                                                -1, -1, -1, -1L)},
+        // make it stream aware
+        true,
+        // enable cross stream sharing?
+        false);
```

Comment:
Is this something we want to make controllable from he API later?

Reply:
It might be. I created it watching the implementation for ROCm EP in `rocm_execution_provider.cc`. It's quite same, if you think this is the situation in ROCm EP implementation then the answer is yes.